internal/test/e2e/fluxGit.go

```diff
@@ -140,6 +141,9 @@ func (e *E2ESession) setupGithubRepo(repo string, envVars map[string]string) (*g
 		return nil, fmt.Errorf("creating repository in Github for test: %v", err)
 	}
 
+	// Wait for eventual consistency in the GitHub API when creating a repository before adding a deploy key
+	time.Sleep(time.Second * 5)
```

Comment:
I can ok this for now, but this is still going to be brittle. It would be better to perform some sort of polling with an eventual limit.